### PR TITLE
D20 convert spectrum axis

### PR DESCRIPTION
**Pull request by gvardany**
**Description of work.**
This PR adds an option to ILL diffraction loader to convert the spectrum axis to 2theta and transpose.
This will make loading/plotting of non detector scan D20 data far more easier for the users.

**To test:**
Pull the newly added unit test file, and load it with:
``` 
ws = LoadILLDiffraction(Filename='170607', ConvertAxisAndTranspose=True)
```
Double click on the workspace, you should see a diffractogram in scattering angle.

---

#### Reviewer ####

**This is a new feature, but since it is small, low risk, yet high impact, it would be great to have in the release.**

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by gui-co — COMMENTED** [review]
I just added two comments/questions about the `#include`s that looked odd to me.
But everything is working fine and the rest looks good to me.

File: Framework/DataHandling/test/LoadILLDiffractionTest.h [hunk 1]
```diff
@@ -9,13 +9,15 @@
...
 #include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/Axis.h"
 #include "MantidAPI/MatrixWorkspace.h"
 #include "MantidAPI/Run.h"
 #include "MantidDataHandling/Load.h"
 #include "MantidDataHandling/LoadILLDiffraction.h"
 #include "MantidGeometry/Instrument/DetectorInfo.h"
 #include "MantidKernel/ConfigService.h"
 #include "MantidKernel/FacilityInfo.h"
+#include "MantidKernel/Unit.h"
```
[the remark this answers is not in the document]
Comment (gvardany) [reply Framework/DataHandling/test/LoadILLDiffractionTest.h:20]: This is needed indeed.

File: Framework/DataHandling/test/LoadILLDiffractionTest.h [hunk 2]
```diff
@@ -9,13 +9,15 @@
 #include <cxxtest/TestSuite.h>
 
 #include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/Axis.h"
```
[the remark this answers is not in the document]
Comment (gvardany) [reply Framework/DataHandling/test/LoadILLDiffractionTest.h:12]: Same here, otherwise it wouldn't compile.

**Review by gui-co — APPROVED** [review]
Everything is good.

**Pull request merged**